bundles/org.openhab.binding.bluetooth.bluegiga/src/main/java/org/openhab/binding/bluetooth/bluegiga/BlueGigaBluetoothDevice.java

```diff
@@ -229,27 +248,18 @@ public void bluegigaEventReceived(BlueGigaResponse event) {
             }
 
             if (connectionState == ConnectionState.DISCOVERING) {
-                // We want to wait for an advertisement and a scan response before we call this discovered.
-                // The intention is to gather a reasonable amount of data about the device given devices send
-                // different data in different packets...
-                // Note that this is possible a bit arbitrary and may be refined later.
                 scanResponses.add(scanEvent.getPacketType());
 
-                if ((scanResponses.contains(ScanResponseType.CONNECTABLE_ADVERTISEMENT)
-                        || scanResponses.contains(ScanResponseType.DISCOVERABLE_ADVERTISEMENT)
-                        || scanResponses.contains(ScanResponseType.NON_CONNECTABLE_ADVERTISEMENT))
-                        && scanResponses.contains(ScanResponseType.SCAN_RESPONSE)) {
```

Comment:
why are you getting rid of these? Aren't these checks important?

Reply:
As mentioned in the first PR:

I removed waiting for a `SCAN_RESPONSE`, as not all devices do send a scan response at all (e.g. EnOcean PTM215B) which completely blocks discovery.  I don't really know for which use case it was added, so I'm not sure what would be a better implementation. I think Bluez has no such thing (because Bluez discovery also works for devices that don't send a SCAN_RESPONSE).

More detailed thoughts on this:

If you have an idea how to follow the original intention of these checks while getting devices that don't send a scan response discovered as well, I'm open to it. Unfortunately, I only have "fire&forget" BLE devices who all do not send a scan response.

I'm not saying it's the best option to remove it, but for now afaik we have not a single binding that makes use of scan response data, all of them check discovery upon manufacturerId or (short) device name.

So by removing it, everything that currently exists will still work, and bindings in the pipe which don't send a scan response will additionally work.

I'm not sure if there will ever be a big need to have the scan response before discovery, but if yes, it'll probably be more a convienience functionality than something which is required for the discovery to properly work. If there is such use case, I think this can still be (re-)implemented in a better way so it'll work for all devices.

Personally, I don't want to come up with an implementation without having a possibility to test this implementation properly with a real device using it.

Reply:
Well you convinced me. Just get rid of the `scanResponses` field too since that now serves no purpose.